uuid.go

```diff
@@ -4,22 +4,36 @@ import (
 	"crypto/rand"
 	"encoding/hex"
 	"fmt"
+	"io"
 )
 
 // GenerateRandomBytes is used to generate random bytes of given size.
 func GenerateRandomBytes(size int) ([]byte, error) {
+	return generateRandomBytesByReader(size, rand.Reader)
+}
+
+func generateRandomBytesByReader(size int, reader io.Reader) ([]byte, error) {
```

Comment:
This should be public and we should be nil checking reader here.

Reply:
It was public, until I suggested it ought to be private.  Why would we have a public function unrelated to uuids in a uuid package?

Reply:
not that this is a reason to make it public, but pointing out that the non-reader specific version `uuid.GenerateRandomBytes(size)` is used throughout Vault.

Reply:
Yeah, I see that now.  nvm, sorry for the noise.